File: find-target-value-sum-in-array-index.js
const getMethod = (array, target) => {
  const map = [];
  const countVal = array.length;
  for (let i = 0; i < countVal; i++) {
    for (let j = i + 1; j < countVal; j++) {
      let sumVal = array[i] + array[j];
      if (sumVal === target) {
        map.push([i, j]);
      }
    }
  }
  return map;
};

const arr = [1, 7, 2, 3, 4, 5, 6];
const targetNum = 11;
const result = getMethod(arr, targetNum);
console.log(result);

File: goalgorithom/find-target-value-sum-in-array-index.go
package main

import "fmt"

var arr = []int{1, 7, 2, 3, 2, 3, 6}
var target = 9
var createMap = make(map[int]int)

func TargetSumValue() {
	//var result = [][]int{}
	for i, value := range arr {
		var current = target - value
		if index, status := createMap[current]; status {
			fmt.Printf("get result %d %d", index, i)
			return
		}
		createMap[value] = i
		//if
	}
	// fmt.Printf("Loop %d", result)
	// fmt.Printf("Get Target vaue %d also get %d %d", target, arr, target)
}
func main() {
	TargetSumValue()
}
